### PR TITLE
[Tests] Add test ids to the runners

### PR DESCRIPTION
**Pull request by fassadlr**
Some tests are failing the build and/or timing out and we have no idea how to identify them. This PR starts to introduce ids passed to the agent string for most of the pos runners.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs [hunk 1]
```diff
@@ -16,11 +16,6 @@ public ProofOfStakeMintCoinsSpecification(ITestOutputHelper outputHelper)
 
         protected override void AfterTest() { }
 
-        private void a_proof_of_stake_node_with_wallet()
```
Comment (zeptin) [inline src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs]: Is this no longer used?

Comment (zeptin) [reply src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs]: I see it isn't

Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs]: It appears not!

**Comment by fassadlr** [comment]
`Staking_Wallet_Can_Mint_New_Coins` was causing the build to hang on occasion so I re-wrote this test the old fashioned way.

File: src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsTest.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+namespace Stratis.Bitcoin.IntegrationTests.Miners
+{
+    public sealed class ProofOfStakeMintCoinsTest
+    {
+        private readonly HashSet<uint256> transactionsBeforeStaking = new HashSet<uint256>();
+        private readonly ConcurrentDictionary<uint256, TransactionData> transactionLookup = new ConcurrentDictionary<uint256, TransactionData>();
+
+        [Fact]
+        public void Staking_Wallet_Can_Mint_New_Coins()
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsTest.cs]: Is this test actually stable?

Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsTest.cs]: Yes, but it had a problem with timing out the build due to it not shutting down the node properly, There was an issue with the BDD style spec so I rewrote it, old school style.

**Pull request merged**